### ISSUE TITLE
only one meter found

### ISSUE THREAD

**Issue by philm142** [unlobito/ha-hildebrandglow#10]
Hi, I've installed the integration and it has found my electricity meter, but not my gas meter, both are linked to my account and I can view them in the bright app. Have you any ideas how I can force it to find my gas meter
Thanks in advance

**Comment by unlobito** [comment]
Hey @philm142, thanks for installing my integration. I'm thrilled to see other people getting use out of it!

I don't have a gas supply at home, so I had to make [some](https://github.com/unlobito/ha-hildebrandglow/blob/7c322a2/custom_components/hildebrandglow/sensor.py#L38) [educated](https://github.com/unlobito/ha-hildebrandglow/blob/7c322a2/custom_components/hildebrandglow/sensor.py#L62) [guesses](https://github.com/unlobito/ha-hildebrandglow/blob/7c322a2/custom_components/hildebrandglow/sensor.py#L72) about how they're exposed by Hildebrand's API. If possible, it'd be great to get some more information from you to understand what's gone wrong here.

First, you'll need to get a session token from Hildebrand's API. If you run the following [curl](https://curl.haxx.se/) command, you'll receive some JSON containing a token element (`"token":  "ey ... UR4"`).

```bash
curl -X POST -H "Content-Type: application/json" \
-H "applicationId: b576fdb0-6e43-4ea4-ac75-a0fd85b0d701" \
-d '{ "username": "your username", "password":"your password" }' \
"https://api.glowmarkt.com/api/v0-1/auth"
```

Once you have the session token, you can place it in the following curl command to see which meters are attached to your account:

```bash
curl -X GET -H "Content-Type: application/json" \
-H "token: ey ... UR4" \
-H "applicationId: b576fdb0-6e43-4ea4-ac75-a0fd85b0d701" \
"https://api.glowmarkt.com/api/v0-1/resource"
```

Are you able to see your gas meter in the returned JSON object? (I'd expect it to be called something like "Smart Meter, gas consumption"). If so, is it labelled with the resourceTypeId `"672b8071-44ff-4f23-bca2-f50c6a3ddd02"`?

If it isn't the same ID, does the integration work after [updating the resourceTypeId expected by GlowConsumptionCurrent](https://github.com/unlobito/ha-hildebrandglow/blob/7c322a2/custom_components/hildebrandglow/sensor.py#L38)?

**Comment by philm142** [comment]
Hi Harley,

sorry for the late reply, I've tried as you've asked and I'm getting:
cannot resolve host errors, do I need to make all four lines into one
command?  sorry my knowledge of using curl is poor!

**Comment by tkerby** [comment]
I suspect here that youre reading the instantaneous electricity use whereas gas is reported every half hour due to the gas meter being battery operated. You might need to read and report the historic data

**Comment by philm142** [comment]
Thanks. Yes the gas meter does only read  every 30 mins. The whole
integration has stopped today, and the entity has stopped, are you having
the same problem. The stick is working and the data is viewable in the
bright app. But the intergration has stopped working

**Comment by tkerby** [comment]
Yeah - the integration stopped working for me a couple of days ago too

**Comment by unlobito** [comment]
Heya,

I reached out to @tkerby over [IRC](https://en.wikipedia.org/wiki/Internet_Relay_Chat) since I recognised his handle from a channel we're both in and was able to get a better understanding of what's gone wrong here.

The component is [currently only requesting the instantaneous usage](https://github.com/unlobito/ha-hildebrandglow/blob/7c322a2/custom_components/hildebrandglow/sensor.py#L106) for each meter detected. As @tkerby explains, this isn't available for gas meters since they're battery powered and can only return usage in ~30min increments as a result. I've reworded this issue's title to better explain the underlying issue and expect the best way to resolve this will be to expose a sensor for gas+electric meters that provides usage for the previous 30min.

The integration will stop working after ~a week because of issue #1, but I have a work-in-progress solution to this on pull request #8. For the time being, you can unbreak the integration by uninstalling it and reinstalling it.

I'm hoping to spend some time this evening getting #8 into a beta state & adding support for past ~30min usage data so gas meters are usable.

Thanks for your patience and for helping me piece together the additional information needed to understand this problem. :)

**Comment by philm142** [comment]
Thank you for your update. I'm grateful for all you hard work and look
forward to the intergration being updated. I'm happy to assist where I can,
but my knowledge is limited.
Thanks Phil

**Comment by ndfred** [comment]
I have a dual energy meter as well, I'll try and give this a go later this week. Would be great indeed to expose electricity and gas, in addition to meter readings as tracked in #13. Thanks for all the hard work @unlobito!

**Comment by dannytsang** [comment]
I'm also on gas and electric meter so if you need any help testing/getting more information let me know.
